src/get-started/first-java-service.md

```diff
@@ -0,0 +1,615 @@
+block:ChangeOwner
+<!--/codeinclude-->
+
+Compile the message:
+
+```shell
+mvn generate-sources
+```
+
+Then navigate to the service `MyService` and add an implementation with
+appropriate constants:
+
+<!--codeinclude-->
+[](../../code-examples/java/exonum-java-binding/tutorials/car-registry/car-registry-service/src/main/java/com/example/car/MyService.java)
+inside_block:ci-change-owner
+<!--/codeinclude-->
+
+This transaction is similar to the first.
+
+Notice how an update of the `owner` field in an existing `Vehicle` value
+is performed. It creates a builder from the existing object with
+`Vehicle.newBuilder(templateVehicle)` method, updates the field, and builds
+a new object.
+
+Compile the code:
+
+```shell
+mvn compile
+```
+
+#### Service Constructor
+
+Finally, we will add a _service constructor_ — a special type of a transaction,
+that is invoked once when the service is instantiated. We will use it
+to populate our registry with some test data.
+
+It is implemented as the `Service#initialize` method. By default, it has
+an empty implementation in the interface, hence it is not yet present
+in `MyService`.
+
+Override the `Service#initialize` with the following implementation:
+
+<!--codeinclude-->
+[](../../code-examples/java/exonum-java-binding/tutorials/car-registry/car-registry-service/src/main/java/com/example/car/MyService.java)
+inside_block:ci-initialize
+<!--/codeinclude-->
+
+Note that this method delegates to the `addVehicle` transaction method
+we have added earlier.
+
+!!! success
+    In this section we have learned how to implement operations modifying
+    the blockchain state: transactions and the service constructor.
+    <!-- todo: "There are more operations of such type: ..." — what is
+    the canonical reference on the topic? Shall we add one? -->
+
+### 4 Service API
+
+In the previous section we have implemented operations modifying the blockchain
+state. Applications usually also need a means to _query_ data. In this section,
+we will add an operation to retrieve a vehicle entry by its ID
+from the registry. This operation will be exposed through REST API.
+
+#### Find Vehicle Service Operation
+
+First, we need to add a query operation to the Service:
+
+<!-- FIXME: Replace lines: selector with inside_block:ci-find-vehicle when
+the bug with braces is resolved: ECR-4318 -->
+<!--codeinclude-->
+[MyService.findVehicle](../../code-examples/java/exonum-java-binding/tutorials/car-registry/car-registry-service/src/main/java/com/example/car/MyService.java)
+lines:138-146
+<!--/codeinclude-->
+
+Although this query method will be invoked by our code, hence the signature
+we use may be arbitrary, the signature is similar to the transaction methods:
+it takes the operation arguments and the context (here: `String` ID
+and `BlockchainData` context).
+
+#### API Controller
+
+<!-- As Exonum does not provide a means to invoke such read operations,
```

Comment:
The Exonum core doesn't provide unified means, but I'd argue that HTTP API is such a means present both in Rust and Java runtime.

Reply:
I thought I shall explain why the API has read requests, but no "submit tx" requests, but commented it out (now removed). We also don't say in this tutorial _how_ the txs get to the node: the user is supposed to just use the provided client (and peek inside, if they are curious). Or shall we explain that (e.g., "the client uses Exonum explorer (?) API to submit txs; and the service API to query its state")?